modules/carousel/jetpack-carousel.php

```diff
@@ -62,10 +66,14 @@ function init() {
 			 * @param array $this->prebuilt_widths Array of default widths.
 			 */
 			$this->prebuilt_widths = apply_filters( 'jp_carousel_widths', $this->prebuilt_widths );
-			add_filter( 'post_gallery', array( $this, 'enqueue_assets' ), 1000, 2 ); // load later than other callbacks hooked it
+			add_filter( 'post_gallery', array( $this, 'check_and_enqueue_assets' ), 1000, 2 ); // load later than other callbacks hooked it
```

Comment:
This filter is being loaded later than others in case some other plugin already processed `[gallery]` shortcode. Meaning that other plugin hooked into the shortcode before Carousel. If so Carousel gallery is not launched - that's the `check_` part of `check_and_enqueue_assets`.